src/read.c

```diff
@@ -3907,6 +3916,14 @@ avifResult avifDecoderReset(avifDecoder * decoder)
             avifDiagnosticsPrintf(&decoder->diag, "Primary item not found");
             return AVIF_RESULT_MISSING_IMAGE_ITEM;
         }
+        int colorItemIndex = -1;
+        for (uint32_t i = 0; i < data->meta->items.count; ++i) {
+            if (colorItem->id == data->meta->items.item[i].id) {
+                colorItemIndex = i;
+                break;
+            }
+        }
+        assert(colorItemIndex >= 0);
```

Comment:
I suggest we move this block of code into `avifDecoderDataFindAlphaItem` so that we don't need to add a new `colorItemIndex` parameter to `avifDecoderDataFindAlphaItem`.

Reply:
Done.

Reply:
Done.